# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My lane and why

I chose **Lane 2: Refresh / Content Opportunity Scoring**.

I want to look at which pages might need attention first instead of trying to improve every page at the same time. The idea is to use the available data to help decide which pages are worth reviewing or refreshing.

I chose this lane because the dataset includes information such as impressions, sessions, clicks, CTR, position, and content age. Looking at these numbers together could help identify pages that may have an opportunity for improvement.

The goal is not to say that a page will definitely perform better after a refresh. Instead, I want to create a useful way to prioritize which pages someone should look at first.


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The question: decision, action, and cost of a wrong call

**Decision:** Which pages should we check first to see if they could benefit from a content refresh?

**Action:** The SEO or content team could use the results to make a list of pages to look at first. After that, they can review the pages and decide what, if anything, should be changed.

**Cost of a wrong call:** If we focus on the wrong pages, we could spend time and effort improving pages that don't really need it. We could also miss pages that actually need more attention.

I want to use the data to make it easier to decide where to start. The results would only be a guide to help with the decision, not a replacement for human judgment.


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/flyrank-bih/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
print("\nColumns:")
print(df.columns.tolist())

Number of rows: 30000
Number of columns: 44

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [2]:
print("Average impressions (90d):", df["impressions_90d"].mean())
print("Average content age (days):", df["content_age_days"].mean())
print("Pages with a downward trend:", (df["trend_direction"] == "down").sum())

Average impressions (90d): 5200.3663
Average content age (days): 256.1678
Pages with a downward trend: 16262


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The dataset has **30,000 pages**. On average, each page got around **5,200 impressions in the last 90 days**, and the average page is about **256 days old**.

What stood out to me most is that **16,262 pages have a downward trend**. That seems like a good reason to look into which pages might need attention or a content refresh.

Because of this, I think this lane is worth exploring further to see if the data can help us decide which pages to review first.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I can and can’t claim

I can use the data to find pages that seem like they might need some attention based on their current performance and trends.

But I can't say for sure that refreshing a page will make it perform better. I also can't claim that the model can predict Google rankings or how Google's algorithm will behave.

For this project, I'll use the results as a guide to help decide which pages to look at first. The final decision will still need to be made by a person after reviewing the page.


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## Self-check

I feel like I have a better idea of what I want to work on now.

* I chose **Lane 2: Refresh / Content Opportunity Scoring**.
* I have a clear question about which pages should be checked first.
* I looked at the data and found some numbers that made this worth exploring.
* I understand that the model is only going to help with prioritizing pages, not give a definite answer.

I think this is a good starting point for the next part of the project.
